libecole/include/ecole/observation/milpbipartite.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <optional>
+
+#include <xtensor/xtensor.hpp>
+
+#include "ecole/observation/abstract.hpp"
+#include "ecole/utility/sparse-matrix.hpp"
+
+namespace ecole::observation {
+
+class MilpBipartiteObs {
+public:
+	using value_type = double;
+
+	static inline std::size_t constexpr n_variable_features = 9;
+	enum struct VariableFeatures : std::size_t {
+		objective = 0,
+		is_type_binary,            // One hot encoded
+		is_type_integer,           // One hot encoded
+		is_type_implicit_integer,  // One hot encoded
+		is_type_continuous,        // One hot encoded
+		has_lower_bound,
+		has_upper_bound,
+        lower_bound,
+        upper_bound,
+	};
+	static inline std::size_t constexpr n_constraint_features = 1;
+    enum struct ConstraintFeatures : std::size_t {
+		bias = 0,
+	};
+
+	xt::xtensor<value_type, 2> variable_features;
+	xt::xtensor<value_type, 2> constraint_features;
+	utility::coo_matrix<value_type> edge_features;
+};
+
+class MilpBipartite : public ObservationFunction<std::optional<MilpBipartiteObs>> {
+public:
+	MilpBipartite(bool use_normalization = false) : normalize{use_normalization}  {}
```

Comment:
Naming is a little bit inconsistent ? I would rather have:
`MilpBipartite(bool normalize_ = false) : normalize{normalize_}  {}`
What is the C++ convention for object properties set in the constructor ?

Reply:
I think I saw Antoine use `m_normalize` for private attributes elsewhere, although I feel it makes the resulting code a bit ugly. What do you think of Maxime's naming suggestion, @AntoinePrv ?

Reply:
We don't have a convention for that. There are some place with the `m_` convention but I agree it's not great.